src/server/rpc/procedures/waterwatch/waterwatch.js

```diff
@@ -0,0 +1,199 @@
+// TODO change console logs.
+// eslint
+let WaterWatchRPC = {
+    isStateless: true,
+    getPath: () => '/waterwatch'
+};
+
+let debug = require('debug'),
+    // request = require('request'),
+    rp = require('request-promise'), //maybe use request-promise-native?
+    error = debug('netsblox:rpc:trends:error'),
+    CacheManager = require('cache-manager'),
+    trace = debug('netsblox:rpc:trends:trace'),
+    Q = require('q');
+
+let baseUrl = 'https://waterservices.usgs.gov/nwis/iv/?',
+    cache = CacheManager.caching({store: 'memory', max: 1000, ttl: 36000});
+
+
+
+// let northernLat = 38;
+// let easternLong = -81;
+// let southernLat = 37;
+// let westernLong = -82;
+
+
+// turn an options object into query friendly string
+function encodeQueryData(options) {
+   let ret = [];
+   for (let d in options)
+     ret.push(encodeURIComponent(d) + '=' + encodeURIComponent(options[d]));
+   return ret.join('&');
+}
+
+// used to send feedback about errors to the caller.
+function showError(err, response) {
+    error('error',err);
+    response.send('Error occured. Bounding box too big?');
+    // response.json(err);
+}
+
+//sanitize coordinate inputs
+
+//******************************
+// factor out message sending and handling of multiple users = sendNext and Stop
+// msgs: an array of msgs stored in global scope.
+let DELAY = 250;
+let msgs = [];
+
+function sendNextMsg(socket){
+    // QUESTION is there a need to check for the socket ID? no need.
```

Comment:
you should check to make sure the msg is intended for the given socket. Here is a problematic scenario:

Suppose you have two sockets, `s1` and `s2` and a message queue with `a1`, `b1`, and `b2` where messages starting with `a` are sent to `s1` and `b` messages are for `s2`. Regardless of who starts sending the messages, this code will fail:

Suppose we start with `s1`:

`s1` sends `a1`
`s2` sends `b1`
`s1` sends `b2`  --- bug!

Starting with `s2` breaks right away with `a1`